heron/healthm/src/java/com/twitter/heron/healthmgr/HealthManager.java

```diff
@@ -0,0 +1,386 @@
+import java.util.logging.Logger;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Singleton;
+import com.google.inject.name.Names;
+import com.microsoft.dhalion.api.IHealthPolicy;
+import com.microsoft.dhalion.api.MetricsProvider;
+import com.microsoft.dhalion.events.EventManager;
+import com.microsoft.dhalion.policy.PoliciesExecutor;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import com.twitter.heron.classification.InterfaceStability.Evolving;
+import com.twitter.heron.classification.InterfaceStability.Unstable;
+import com.twitter.heron.common.utils.logging.LoggingHelper;
+import com.twitter.heron.healthmgr.common.HealthMgrConstants;
+import com.twitter.heron.healthmgr.common.PackingPlanProvider;
+import com.twitter.heron.healthmgr.sensors.TrackerMetricsProvider;
+import com.twitter.heron.scheduler.client.ISchedulerClient;
+import com.twitter.heron.scheduler.client.SchedulerClientFactory;
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.ConfigLoader;
+import com.twitter.heron.spi.common.Context;
+import com.twitter.heron.spi.common.Key;
+import com.twitter.heron.spi.statemgr.IStateManager;
+import com.twitter.heron.spi.statemgr.SchedulerStateManagerAdaptor;
+import com.twitter.heron.spi.utils.ReflectionUtils;
+
+/**
+ * {@link HealthManager} makes a topology dynamic and self-regulating. This is implemented using
+ * Dhalion library. The {@link HealthManager} will perform the following functions to achieve its
+ * goal:
+ * <ul>
+ * <li>loads heron configuration including health policy configuration from
+ * <code>healthmgr.yaml</code>
+ * <li>initializing guice injector with metrics collection module from <code>tracker</code> or
+ * <code>metrics cache</code>, <code>scheduler client</code> and <code>state client</code>
+ * <li>initializes health policies instances and starts policy execution using
+ * {@link PoliciesExecutor}
+ * </ul>
+ * The {@link HealthManager} is executed as a process. It is recommended that it is started on
+ * container 0, colocated with the metrics provider and the scheduler service.
+ * <p>
+ * Required command line options for the {@link HealthManager} include
+ * <ul>
+ * <li>cluster name: <code>-c local</code>
+ * <li>role: <code> -r dev</code>
+ * <li>environment: <code> -e default</code>
+ * <li>heron home directory: <code> -d ~/.heron</code>
+ * <li>config directory: <code> -p ~/.heron/conf</code>
+ * <li>topology name: <code> -n AckingTopology</code>
+ * </ul>
+ * <p>
+ * Optional command line options for the {@link HealthManager} include
```

Comment:
Yes. it will be part of a new PR. User would provide the dry run configuration using policy config file. Dhalion policy executor will just log the diagnosis and skip execution of resolver if dry run is true.